### PR TITLE
Add badges for versions and CI

### PR DESCRIPTION
**Pull request by 76creates**
Badges make for good user experience so we should add a badge for release version, preview version, and CI status.
Added `preview version` section to the README.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,8 @@
 # ILGPU
+![CI](https://github.com/G-Research/consuldotnet/workflows/CI/badge.svg?style=flat)
```
Comment (m4rs-mt) [inline README.md:2]: I think it would be better to import a CI-badge instead of referring to another repository for this purpose. What do you think?

**Review by m4rs-mt — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (76creates) [reply README.md:2]: heh, happens to the best of us 😅

File: README.md [hunk 2]
```diff
@@ -1,4 +1,8 @@
 # ILGPU
+![CI](https://github.com/m4rs-mt/ILGPU/actions/workflows/ci.yml/badge.svg?style=flat)
```
Comment (jgiannuzzi) [inline README.md:2]: Maybe we want to refine our badge to link to the workflow runs and be only for `push` events to `master`?

```suggestion
[![CI](https://github.com/m4rs-mt/ILGPU/actions/workflows/ci.yml/badge.svg?branch=master&event=push)](https://github.com/m4rs-mt/ILGPU/actions/workflows/ci.yml)
```

File: README.md [hunk 3]
```diff
@@ -1,4 +1,8 @@
...
+[![](https://img.shields.io/nuget/v/ilgpu?style=flat&label=release)](https://www.nuget.org/packages/ilgpu/)
+[![](https://img.shields.io/nuget/vpre/ilgpu?style=flat&label=pre-release)](https://www.nuget.org/packages/ilgpu/)
```
Comment (jgiannuzzi) [inline README.md:4]: I would suggest we link to the corresponding releases like this:
```suggestion
[![](https://img.shields.io/nuget/v/ilgpu?style=flat&label=release)](https://www.nuget.org/packages/ilgpu/latest)
[![](https://img.shields.io/nuget/vpre/ilgpu?style=flat&label=pre-release)](https://www.nuget.org/packages/ilgpu/absoluteLatest)
```

File: README.md (hunk 2 above)
Comment (76creates) [reply README.md:2]: > Maybe we want to refine our badge to link to the workflow runs and be only for `push` events to `master`?

sounds very good, wasn't aware of that possibility ⚡

Comment (jgiannuzzi) [reply README.md:2]: actually the link to the workflow runs should also be filtered for `push` events to `master` only: https://github.com/m4rs-mt/ILGPU/actions/workflows/ci.yml?query=branch%3Amaster+event%3Apush

**Review by jgiannuzzi — APPROVED** [review]

**Review by m4rs-mt — APPROVED** [review]

**Pull request merged**